Neos.Media/Classes/Domain/Model/AssetSource/AssetTypeFilter.php

```diff
@@ -0,0 +1,57 @@
+<?php
+namespace Neos\Media\Domain\Model\AssetSource;
+
+/*
+ * This file is part of the Neos.Media package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+  *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+final class AssetTypeFilter implements \JsonSerializable
+{
+    /**
+     * @var string
+     */
+    private $assetType = 'All';
+
+    /**
+     * AssetType constructor.
+     *
+     * @param string $assetType
+     */
+    public function __construct(string $assetType)
+    {
+        if (!in_array($assetType, ['All', 'Image', 'Document', 'Video', 'Audio'])) {
+            throw new \InvalidArgumentException('Invalid asset type.', 1511797661398);
```

Comment:
1511797661398 is too long for a timestamp / includes millis

Reply:
That's taken from some existing code, but I'll fix it.